src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -83,6 +87,22 @@ def _pull_image(image_str, job_dir):
                              f"Standard error: {std_err.read()}")
 
 
+def _write_to_file(file_path, content, mode="x", ignore_existing=False):
+    """
+    Writes the specified content to the specified file.
+
+    If the file already exists, an exception is raised unless ignore_existing is set to True.
+    """
+    try:
+        with open(file_path, mode) as f:
+            f.write(content)
+    except FileExistsError:
+        if ignore_existing:
+            print(f"File {file_path} already exists.")
+        else:
+            raise FileExistsError(f"File {file_path} already exists.")
```

Comment:
```suggestion
            raise
```
any reason not to use the same exception?

Reply:
👍

Reply:
:+1: